packages/ws/src/utils/constants.ts

```diff
@@ -18,13 +18,19 @@ export enum Encoding {
  * Valid compression methods
  */
 export enum CompressionMethod {
-	ZlibStream = 'zlib-stream',
+	ZLibNative,
+	ZLibSync,
 }
 
 export const DefaultDeviceProperty = `@discordjs/ws [VI]{{inject}}[/VI]` as `@discordjs/ws ${string}`;
 
 const getDefaultSessionStore = lazy(() => new Collection<number, SessionInfo | null>());
 
+export const CompressionParameterMap = {
+	[CompressionMethod.ZLibNative]: 'zlib-stream',
+	[CompressionMethod.ZLibSync]: 'zlib-stream',
+} as const satisfies Record<CompressionMethod, string>;
```

Comment:
Isn't this a `Readonly<Record<...>>`?

Reply:
`satisfies` does not alter the type of the variable, it is merely used for validating that the object conforms to that type. Besides, `as const` already makes the type readonly.

Reply:
What I meant was wouldn't it be more ideal to satisfy something that's `as const` (which is immutable) with something that is not immutable? So it should be like:

```suggestion
} as const satisfies Readonly<Record<CompressionMethod, string>>;
```

Reply:
it's not any better IMO, just more verbose for no good reason

Reply:
+1, the exported type is still going to be marked as readonly